packages/js/product-editor/src/blocks/section/edit.tsx

```diff
@@ -15,28 +17,91 @@ import {
  */
 import { sanitizeHTML } from '../../utils/sanitize-html';
 import { SectionBlockAttributes } from './types';
+import { hasAttributesUsedForVariations } from '../../utils';
 
 export function Edit( {
 	attributes,
-}: BlockEditProps< SectionBlockAttributes > ) {
+	context,
+}: BlockEditProps< SectionBlockAttributes > & {
+	context?: {
+		selectedTab?: string | null;
+	};
+} ) {
 	const { description, title, blockGap } = attributes;
+	const tab = context?.selectedTab || '';
 	const blockProps = useBlockProps();
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+	const isDisabledSection =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		[ 'inventory', 'pricing', 'shipping' ].includes( tab );
 	const innerBlockProps = useInnerBlocksProps(
 		{
 			className: classNames(
 				'wp-block-woocommerce-product-section__content',
-				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`
+				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`,
+				{
+					'is-disabled-section': isDisabledSection,
+				}
 			),
 		},
 		{ templateLock: 'all' }
 	);
 	const SectionTagName = title ? 'fieldset' : 'div';
 	const HeadingTagName = SectionTagName === 'fieldset' ? 'legend' : 'div';
 
+	useEffect( () => {
+		const elementsToReEnable: (
+			| HTMLInputElement
+			| HTMLButtonElement
+			| HTMLSelectElement
+			| HTMLTextAreaElement
+		 )[] = [];
+
+		if ( isDisabledSection ) {
+			const disabledSections = document.querySelectorAll(
+				'.is-disabled-section'
+			);
+
+			disabledSections.forEach( ( section ) => {
+				const controls = section.querySelectorAll(
+					'input, button, select, textarea'
+				);
```

Comment:
This is implementation knowledge that the section shouldn't know about.

If we were continue with this approach (keeping the logic in the general section block), I still think it would be better for blocks to have a `disabled` attribute that we could toggle. That way, it is up to the block to render itself as disabled.

Reply:
For this particular solution, I believe that the blocks section is the most suitable place for the code that disables the section's forms. I'm not entirely clear about the 'disabled' attribute. Using it would require every block to verify whether we created a variation. With this interim solution (which we plan to modify after implementing the templates solution we're currently developing), we only need to check that condition once.
Did I perhaps misunderstand your suggestion?